### PR TITLE
Remove deprecated variant of getRegistrationAssertion.

### PR DESCRIPTION
**Pull request by kaniyan**
Remove deprecated variant of `AbstractProtocolAdapterBase.getRegistrationAssertion` method and its usage.

### CODE REVIEW

File: service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java [hunk 1]
```diff
@@ -370,7 +370,7 @@ public void testGetRegistrationAssertionSucceedsForExistingDevice(final VertxTes
 
         final Checkpoint assertion = ctx.checkpoint();
         // WHEN an assertion for the device is retrieved
-        adapter.getRegistrationAssertion("tenant", "device", null)
+        adapter.getRegistrationAssertion("tenant", "device", null, mock(SpanContext.class))
```
Comment (sophokles73) [inline service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java]: this method is being invoked intentionally here. We are testing both variants, the deprecated one, and the not-deprecated one (few lines below) ...

File: service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java [hunk 2]
```diff
@@ -401,7 +401,7 @@ public void testGetRegistrationAssertionFailsWith404ForNonExistingDevice(final V
 
         final Checkpoint assertion = ctx.checkpoint();
         // WHEN an assertion for a non-existing device is retrieved
-        adapter.getRegistrationAssertion("tenant", "non-existent", null)
+        adapter.getRegistrationAssertion("tenant", "non-existent", null, mock(SpanContext.class))
```
Comment (sophokles73) [inline service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java]: same here
as long as we have the deprecated methods in the `RegistrationClient`, we will keep this test, I'd say ...

Comment (kaniyan) [reply service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java]: `RegistrationClient` has no deprecated methods. Do you mean `AbstractProtocolAdapterBase`? yes, we can remove the usage in the above tests, when we remove the deprecated method from  `AbstractProtocolAdapterBase`.

**Comment by sophokles73** [comment]
@kaniyan can you extend this PR with the removal of the deprecated
`AbstractProtocolAdapterBase.getRegistrationAsertion()` method?
I think the time is right for it now. We can then merge this PR.

**Comment by kaniyan** [comment]
> @kaniyan can you extend this PR with the removal of the deprecated
> `AbstractProtocolAdapterBase.getRegistrationAsertion()` method?
> I think the time is right for it now. We can then merge this PR.

@sophokles73 Done :-)

File: site/homepage/content/release-notes.md [hunk 3]
```diff
@@ -7,6 +7,9 @@ title = "Release Notes"
 ### API Changes
 * The obsolete method variants of `reportTelemetry` in `org.eclipse.hono.service.metric.Metrics` 
   have been removed. The new variants of this method accept an additional parameter of type `TenantObject`.
+* The already deprecated `org.eclipse.hono.service.AbstractProtocolAdapterBase.getRegistrationAssertion`
+  method has been removed. The alternate variant of the `getRegistrationAssertion` method which accepts an 
+  additional OpenTracing span parameter should be used.
```
Comment (sophokles73) [inline site/homepage/content/release-notes.md]: :+1:

**Pull request merged**